src/commit_list.c

```diff
@@ -107,75 +110,32 @@ git_commit_list_node *git_commit_list_pop(git_commit_list **stack)
 
 static int commit_quick_parse(
 	git_revwalk *walk,
-	git_commit_list_node *commit,
-	const uint8_t *buffer,
-	size_t buffer_len)
+	git_commit_list_node *node,
+	git_odb_object *obj)
 {
-	const size_t parent_len = strlen("parent ") + GIT_OID_HEXSZ + 1;
-	const uint8_t *buffer_end = buffer + buffer_len;
-	const uint8_t *parents_start, *committer_start;
-	int i, parents = 0;
-	int64_t commit_time;
-
-	buffer += strlen("tree ") + GIT_OID_HEXSZ + 1;
-
-	parents_start = buffer;
-	while (buffer + parent_len < buffer_end && memcmp(buffer, "parent ", strlen("parent ")) == 0) {
-		parents++;
-		buffer += parent_len;
-	}
-
-	commit->parents = alloc_parents(walk, commit, parents);
-	GITERR_CHECK_ALLOC(commit->parents);
+	int error;
+	size_t i;
+	git_oid *parent_oid;
+	git_commit *commit = git__calloc(1, git_odb_object_size(obj));
+	commit->object.repo = walk->repo;
```

Comment:
Note : that was what my old `__init_from_odb` function was about. Right now I'm cheating by providing only what's necessary for the parser to work (access to the repo for applying the grafts). I can either bring back the `__init_from_odb` function, or make the parser not crash if there's no repo around. Suggestions welcome !